Cuphead/Cuphead.asl

```diff
@@ -300,17 +346,26 @@ split
 
 reset
 {
-	if (current.InILMode && (current.Loading && current.Time == 0f || current.InOverworld))
+	if (current.InILMode && (current.Loading && current.LSDTime == 0f || current.InOverworld))
 	{
-		vars.Log("Resetting due to reset {} | Time: " + current.Time + " | Loading: " + current.Loading + " | InOverworld: " + current.InOverworld);
+		vars.Log("Resetting due to reset {} | Time: " + current.Time + " | InOverworld: " + current.InOverworld + " | InKingDice: " + current.InKingDice + " | Loading: " + current.Loading);
+		current.IsKDLevelEnding = false;
 		return true;
 	}
 }
 
 gameTime
 {
 	if (current.InILMode)
-		return TimeSpan.FromSeconds(current.Time);
+	{
+		if (!current.InKingDice)
+			return TimeSpan.FromSeconds(current.Time);
+		
+		// King dice is a series of levels whose time at the end is a sum of all levels, updated after each level is completed.
+		// If a level is ending we just use that time, otherwise we update the current time with the current level time
+		var time = current.IsKDLevelEnding ? current.LSDTime : current.LSDTime + current.Time;
+		return TimeSpan.FromSeconds(time);
+	}
```

Comment:
Return early;
```cs
if (current.InILMode)
	return;

if (!current.InKingDice)
	return TimeSpan.FromSeconds(current.Time);
		
// King dice is a series of levels whose time at the end is a sum of all levels, updated after each level is completed.
// If a level is ending we just use that time, otherwise we update the current time with the current level time
var time = current.IsKDLevelEnding ? current.LSDTime : current.LSDTime + current.Time;
return TimeSpan.FromSeconds(time);
```